src/main/java/org/jboss/msc/SimpleService.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.msc;
+
+import org.jboss.msc.service.StartContext;
+import org.jboss.msc.service.StopContext;
+
+import java.util.function.Consumer;
+
+/**
+ * Simple service implementation which provides just single value.
+ *
+ * Sample usage:
+ * <pre>
+ *     ServiceTarget target = ...
+ *     ServiceName simpleServiceName = ...
+ *     ServiceName providedValueName = ...
+ *     ServiceBuilder&lt;String&gt; sb = target.addService(simpleServiceName);
+ *     sb.setInstance(new SimpleService<>(sb.provides(providedValueName), new String("injected Value")));
+ *     sb.install();
+ * </pre>
+ *
+ * @author <a href="mailto:ropalka@redhat.com">Richard Opalka</a>
+ * @param <V> provided value type
+ */
+public final class SimpleService<V> implements Service {
```

Comment:
I have one comment here.  This class should probably be hidden as package-private and constructed with a factory method like this:

```java
Service myService = Service.createSimpleService(injector, value);
```

Then we could potentially expand this API to work with ``Supplier``, ``Function``, etc. later on without having another class to deprecate.